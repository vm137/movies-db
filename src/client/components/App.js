import React from 'react';
import './App.scss';
import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
import Header from './Header/Header';
import SearchBlock from './SearchBlock/SearchBlock';

const App = () => {
    return (
      <div className='wrapper'>
        <Header />
        <ErrorBoundary>
          <SearchBlock />
        </ErrorBoundary>

        <div className="push" />
      </div>
    )
};

export default App;
