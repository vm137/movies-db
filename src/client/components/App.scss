@import '../constants/constants';

html, body {
    height: 100%;
}

html, body, p {
    margin: 0;
}

.wrapper {
    position: relative;
    width: 900px;
    margin: 0 auto;
}

#app {
    min-height: 100%;
    margin-bottom: -50px;
}

.footer, .push {
    height: 50px;
}

.footer {
    position: relative;
    width: 900px;
    margin: 0 auto;
    background-color: $BACKGROUND_BLUE;

    .logo-bottom {

    }
}
