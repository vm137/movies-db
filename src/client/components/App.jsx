import React from 'react';
import './App.scss';
import ErrorBoundary from './ErrorBoundary/ErrorBoundary';
import MainWindow from './MainWindow/MainWindow';

const App = () => {
    return (
      <div className='wrapper'>
        <ErrorBoundary>
          <MainWindow />
        </ErrorBoundary>

        <div className="push" />
      </div>
    )
};

export default App;
