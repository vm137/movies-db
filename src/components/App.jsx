import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import './App.scss';
import ErrorBoundary from './ErrorBoundary';
import MainWindow from './MainWindow';
import store from '../store';

// eslint-disable-next-line react/prop-types
const App = ({ Router, location, context }) => (
  <div className="wrapper">
    <ErrorBoundary>
      <Provider store={store}>
        <MainWindow Router={Router} location={location} context={context} />
      </Provider>
    </ErrorBoundary>
    <div className="push" />
  </div>
);

App.prototypes = {
  Router: PropTypes.object.isRequired,
};

export default App;
