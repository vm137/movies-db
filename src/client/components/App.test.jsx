import { shallow } from 'enzyme';
import React from 'react';
import App from './App';

describe('<App />', () => {
  it('renders matching snapshot', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('has .wrapper class', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.wrapper').length).toBe(1);
  });

  it('renders matching wrapper', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.push').length).toBe(1);
  });
});
