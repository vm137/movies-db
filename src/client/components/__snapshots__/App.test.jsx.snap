// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders matching snapshot 1`] = `
<div
  className="wrapper"
>
  <ErrorBoundary>
    <MainWindow />
  </ErrorBoundary>
  <div
    className="push"
  />
</div>
`;
